model/jpa/src/main/java/org/keycloak/models/jpa/session/JpaUserSessionPersisterProvider.java

```diff
@@ -375,10 +381,13 @@ public UserSessionModel loadUserSession(RealmModel realm, String userSessionId,
 
             closing(queryClientSessions.getResultStream()).forEach(clientSession -> {
                 PersistentUserSessionAdapter userSession = sessionsById.get(clientSession.getUserSessionId());
-                boolean added = addClientSessionToAuthenticatedClientSessionsIfPresent(userSession, clientSession);
-                if (!added) {
-                    // client was removed in the meantime
-                    removedClientUUIDs.add(clientSession.getClientId());
+                // check if we have a user session for the client session
+                if (userSession != null) {
```

Comment:
Should we remove clientSession if userSession doesn't exist?

Reply:
Not sure. We would probably need to introduce a query for that and still we don't have a way to delete it from a cache if it exists there.

Reply:
This is a good point, but not in this PR.

Reply:
I see. Ok then, I will approve